packages/desktop-client/src/hooks/usePreviewTransactions.ts

```diff
@@ -1,34 +1,66 @@
-import { useMemo } from 'react';
+import { useState } from 'react';
 
 import {
   type ScheduleStatuses,
   useCachedSchedules,
 } from 'loot-core/client/data-hooks/schedules';
+import { send } from 'loot-core/platform/client/fetch';
+import { ungroupTransactions } from 'loot-core/shared/transactions';
 import { type ScheduleEntity } from 'loot-core/types/models';
 
-export function usePreviewTransactions() {
+import { type TransactionEntity } from '../../../loot-core/src/types/models/transaction.d';
+
+export function usePreviewTransactions(
+  collapseTransactions: (ids: string[]) => void,
```

Comment:
Is it possible to call this outside the hook? Calling it here seems out of place since the hook is just intended to retrieve the preview transactions.

Reply:
Yeah, it's a bit out of place for sure. I went with this approach because the logic needs to run when the preview transactions are _loaded_ rather than in a render somewhere (I tried an approach with a `useEffect` before this and it got extremely messy). One mitigation maybe is to rename from `collapseTransactions` to `onLoad`. What do you think?